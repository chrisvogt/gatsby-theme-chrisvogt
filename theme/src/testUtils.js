import React from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { ThemeProvider } from 'theme-ui'
import theme from './gatsby-plugin-theme-ui'

import store from './store'

export { store }

// React module that provides necessary context to components being tested.
export const TestProvider = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>

export const TestProviderWithState = ({ children }) => (
  <ReduxProvider store={store}>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </ReduxProvider>
)
