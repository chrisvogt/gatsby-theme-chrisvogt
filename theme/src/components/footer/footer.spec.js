import React from 'react'
import renderer from 'react-test-renderer'

import Content from './content'
import Footer from './footer'
import { TestProvider } from '../../testUtils'
import useSiteMetadata from '../../hooks/use-site-metadata'

jest.mock('./content')
jest.mock('../../hooks/use-site-metadata')

describe('Footer', () => {
  Content.mockImplementation(() => <div className='MOCK__FooterContent'></div>)
  const mockFooterText = 'Â© 2019-2020 Chris Vogt'

  useSiteMetadata.mockImplementation(() => ({
    footerText: mockFooterText
  }))

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <TestProvider>
          <Footer />
        </TestProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
