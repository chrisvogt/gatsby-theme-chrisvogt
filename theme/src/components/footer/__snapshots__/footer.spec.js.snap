// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches the snapshot 1`] = `
<div
  className="css-1jzzxh0"
>
  <svg
    className="css-1hgawz4"
    preserveAspectRatio="xMidYMax slice"
    viewBox="0 0 1366 64"
    width="100%"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1366,0V60.42c-.2,2.32-88.53,7-218.38-.43-129.74-6.07-301-24.16-464.62-36.87C519.8,9.31,348.92,6,218.94,19.73,89,31.35-.15,60,0,61.54V0Z"
      fill="#fcfcfc"
    />
  </svg>
  <div
    className="MOCK__FooterContent"
  />
</div>
`;
